libraries/AP_NavEKF3/AP_NavEKF3_Control.cpp

```diff
@@ -461,35 +477,54 @@ bool NavEKF3_core::readyToUseBodyOdm(void) const
 // return true if the filter to be ready to use gps
 bool NavEKF3_core::readyToUseGPS(void) const
 {
-    return validOrigin && tiltAlignComplete && yawAlignComplete && (delAngBiasLearned || assume_zero_sideslip()) && gpsGoodToAlign && (frontend->_fusionModeGPS != 3) && gpsDataToFuse && !gpsInhibit;
+    if (frontend->sources.getPosXYSource() != AP_NavEKF_Source::SourceXY::GPS) {
+        return false;
+    }
+
+    return validOrigin && tiltAlignComplete && yawAlignComplete && (delAngBiasLearned || assume_zero_sideslip()) && gpsGoodToAlign && gpsDataToFuse && !gpsInhibit;
```

Comment:
I don't think the order of the checks should make any difference.  I've just added a check at the top of the function so that we don't ever consider the source "ready" unless the source has become active (i.e. the user has configured it through the SRC_ parameters and the active source set includes the source).

Reply:
we discussed this and I think we agreed that the compute costs are low enough that this is OK.